SECURITY.md

```diff
@@ -256,6 +256,14 @@ embargo policy to share binary distributions of the security fixes before the pu
 This includes, but is not limited to, Envoy binaries and Docker images. It is expected that
 distributors have a method to stage and validate new binaries without exposing them publicly.
 
+If Envoy is running in a hosted environment, the fix may rollout ahead of the embargo lift date
+only if the end user **cannot** access the running Envoy process and possibly reverse engineer the fix.
+If a security fix includes changes to the controlplane then the controlplane binaries, Docker images,
+and running process should not be accessible to the end user. If the above conditions hold in a hosted
+controlplane then the rollout may proceed ahead of the embargo lift date. If the controlplane produces
+XDS output containing textual keys pertaining to the fix, such as specific filter metadata keys then
+the hosted controlplane **may not** be rolled out until the embargo lifts.
```

Comment:
This seems somewhat strict; I think the answer depends on whether the xDS is visible to end users. If a hosted control plane is programming a hosted data plane and the xDS is hidden and admin config dump is disabled, this seems a valid situation to do embargoed rollouts.

Reply:
The wording is a bit confusing, but I believe that the intent behind "textual keys" in "textual keys pertaining to the fix, such as specific filter metadata keys" is that embargoed rollout is forbidden only if there are new fields/keys/values that could be leaked to the data path via Lua/Wasm plugins, access logs, etc., and not if there are any new protobuf fields in xDS, since that would be always true for a control plane that includes security fixes.

I agree that embargoed rollouts should be conditioned on config dump being unavailable to the users. @mandarjog could you add this?